pkg/agent/client.go

```diff
@@ -293,12 +293,17 @@ func (a *Client) Recv() (*client.Packet, error) {
 	a.recvLock.Lock()
 	defer a.recvLock.Unlock()
 
+	const segment = commonmetrics.SegmentToAgent
 	pkt, err := a.stream.Recv()
 	if err != nil && err != io.EOF {
 		metrics.Metrics.ObserveServerFailureDeprecated(metrics.DirectionFromServer)
-		metrics.Metrics.ObserveStreamErrorNoPacket(commonmetrics.SegmentToAgent, err)
+		metrics.Metrics.ObserveStreamErrorNoPacket(segment, err)
 	}
-	return pkt, err
+	if err != nil {
+		return nil, err
+	}
+	metrics.Metrics.ObservePacket(segment, pkt.Type)
```

Comment:
I think this is double counting with the observe on line 398?

Reply:
Thanks, fixed.